### ISSUE TITLE
Unable to open database file

### ISSUE THREAD

**Issue by AliOsm** [Shipyrd/shipyrd#27]
Thanks for this great work!

When I followed the documentation, I faced the following error:

```
ActiveRecord::StatementInvalid: SQLite3::CantOpenException: unable to open database file (ActiveRecord::StatementInvalid)
```

Also, I can't see the volume of the docker container. Any thoughts?

**Comment by AliOsm** [comment]
Ok I found the issue. I used volume path as:

```
data:/rails/db/production
```

But on my server, I should use:

```
/root/shipyrd:/rails/db/production
```

I think this is a good thing to add to the documentation.

**Comment by AliOsm** [comment]
Also I used the following labels to have HTTPS setup with cloudflare and letsencrypt:

```
traefik.http.routers.baheth-shipyrd.entrypoints: websecure
traefik.http.routers.baheth-shipyrd.tls: true
traefik.http.routers.baheth-shipyrd.tls.certresolver: letsencrypt
```

**Comment by AliOsm** [comment]
Finally, to be able to access the DB and write to it I should ran:

```
sudo groupadd ubuntu
sudo useradd -m -g ubuntu -s /bin/bash ubuntu
sudo chown -R ubuntu:ubuntu shipyrd
```

In a pre-connect script. Otherwise, the Shipyrd application wasn't able to write to the DB.

**Comment by nickhammond** [comment]
@AliOsm I was previously using litestack for the sqlite access which resolved this issue, looks like the sqlite3 gem doesn't handle this as well which requires setting the additional permissions. I'll most likely switch back to litestack for the sqlite access to resolve this since it doesn't require setting permissions and utilizes a docker volume.

**Comment by AliOsm** [comment]
You are not switching to psql right? Because I can see a pg branch :3

**Comment by nickhammond** [comment]
Haha no, that was a while back. Sqlite definitely since it's easier to ship around and less dependencies. psql was when I was testing out hotwire over psql which doesn't work very well.

**Comment by nickhammond** [comment]
@AliOsm This should be fixed in [v0.3.5](https://github.com/Shipyrd/shipyrd/releases/tag/v0.3.5) if you want to give it a try. Permissions are now set within the _Dockerfile_ for the sqlite DB directory(_/rails/db/production_). This is intended to work with the Docker volume style mentioned in the readme `data:/rails/db/production` or `shipyrd:/rails/db/production`.

**Comment by nickhammond** [comment]
@AliOsm Added a note about upgrading to the readme https://github.com/Shipyrd/shipyrd?tab=readme-ov-file#upgrading
